client/src/components/Search/index.js

```diff
@@ -3,7 +3,7 @@ import style from './style';
 import axios from 'axios';
 import {API_SERVER} from '../../../config';
 const OK_Status = 'OK';
-
+let marker;
```

Comment:
Unless I'm missing something, let's make this a state variable of the search component instead of a global variable for the module. In the constructor, declare in the state object and initialize as null. I'll annotate the rest of the code where related changes are needed.